scrapy/czlscrape/spiders/cultura.py

```diff
@@ -0,0 +1,179 @@
+    (r"\b(ordonanta de guvern|og)\b", "OG"),
+    (r"\b(ordonanta de urgenta|oug)\b", "OUG"),
+    (r"\b(ordin de ministru|ordin|ordinul|om)\b", "OM")
+]
+INITIATIVE_TEXT_CLEANUP_PATTERNS = [r'(Fisiere|Fișiere|Anexe)[^:]*:']
+INITIATIVE_TEXT_CLEANUP_HTML_TAGS = ['ul', 'a']
+DIACRITICS_RULES = [
+    (r'[șş]', 's'),
+    (r'[ȘŞ]', 'S'),
+    (r'[țţ]', 't'),
+    (r'[ȚŢ]', 'T'),
+    (r'[ăâ]', 'a'),
+    (r'[ĂÂ]', 'A'),
+    (r'[î]', 'i'),
+    (r'[Î]', 'I')
+]
+
+
+def remove_html_tags(selector: Selector, tags: list) -> Selector:
+    dom_element = selector.root
+    for tag in tags:
+        html_tag_selector = selector.xpath(tag)
+        if html_tag_selector:
+            dom_element.remove(html_tag_selector[0].root)
+    return selector
+
+
+def extract_text(selector: Selector) -> str:
+    return selector.xpath('string(.)').extract_first().strip()
+
+
+def collapse_blank_lines(text: str) -> str:
+    lines = re.sub("\n\s*\n{2,}", "\n\n", text).split("\n\n")
+    return "\n\n".join(line.strip() for line in lines)
+
+
+def remove_diacritics(text: str) -> str:
+    result = text
+    for search_pattern, replacement in DIACRITICS_RULES:
+        result = re.sub(search_pattern, replacement, result)
+    return result
+
+
+def guess_initiative_type(text: str) -> str:
+    text_without_diacritics = remove_diacritics(text)
+    for search_pattern, publication_type in INITIATIVE_TYPE_RULES:
+        if re.search(search_pattern, text_without_diacritics, re.IGNORECASE):
+            return publication_type
+    else:
+        return "OTHER"
+
+
+def cleanup_article_text(selector: Selector) -> str:
+    selector_copy = remove_html_tags(selector, INITIATIVE_TEXT_CLEANUP_HTML_TAGS)
+    text = extract_text(selector_copy)
+    for cleanup_pattern in INITIATIVE_TEXT_CLEANUP_PATTERNS:
+        text = re.sub(cleanup_pattern, '', text)
+    text = collapse_blank_lines(text)
+    return text
+
+
+def extract_publish_date_components(selector: Selector) -> tuple:
+    date_pattern_match = re.match(PUBLISH_DATE_PATTERN, extract_text(selector))
+    if date_pattern_match:
+        return int(date_pattern_match.group('day')), romanian_month_number(date_pattern_match.group('month')), int(date_pattern_match.group('year'))
+
+
+def romanian_month_number(text: str) -> int:
+    return ROMANIAN_MONTHS.get(text.lower())
+
+
+def format_date(day: int, month: int, year: int) -> str:
+    return "{0}-{1:02d}-{2:02d}".format(year, month, day)
+
+
+class CulturaSpider(Spider):
+    name = 'cultura'
+    start_urls = [INDEX_URL]
+
+    def parse(self, response):
+        for article in response.css('#recomended-articles > div > div > div > div.recommended-title'):
+            href = article.css('a::attr(href)').extract_first()
+            yield Request(response.urljoin(href), self.parse_article)
+
+    def parse_article(self, response):
+        now = datetime.datetime.now()
+        current_day, current_month, current_year = now.day, now.month, now.year
```

Comment:
Got it. Am scris asta cand erau folosite in 2 ocazii diferite, dar apoi am sters un caz. Rezolv.